### ISSUE TITLE
[Desktop] Manual/Fine Page Zoom Control

### ISSUE THREAD

**Issue by scudder12** [brave/brave-browser#10353]
## Description 

Currently when zooming a tab/page, one can only go in fixed increments, e.g. 80%, 90%, etc. Given that so many sites now use various font sizes and page spacing that may or may not be optimal for desktop browsing (since everyone seems to be assuming most users are on a mobile device these days) it would be nice to have finer zoom control, down to single percentage points, ideally. Seems like having the + and - buttons adjust in increments as they do now, but allowing manual entry of a number in between the preset increments would be a good implementation that would keep it straightforward but customizable when needed. 

## Steps to Reproduce

   1. When clicking the + and - buttons zooming only occurs in preset increments.

## Actual result:

Zoom levels are limited to the preset increments, e.g. 90%, 100%, 110%, etc.

## Expected result:
The + and - buttons would zoom up or down to the next preset increment as normal, but the text box showing the current zoom level would be editable, allowing for entry of any desired zoom value in whole percentage points, e.g. 94%. When clicking the + or - buttons after setting a custom zoom setting, the zoom level would simply move up or down to the next nearest preset level (e.g. from a custom setting of 94%, pressing the + button would zoom to 100% as that is the next highest preset level).

## Reproduces how often: 

Easily reproduced.

## Brave version (brave://version info)

Brave | 1.10.93 Chromium: 83.0.4103.106 (Official Build) (64-bit)
-- | --
Revision | ce7134bb3d95141cd18f1e65772a4247f282d950-refs/branch-heads/4103@{#694}
OS | macOS Version 10.15.5 (Build 19F101)

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No
- Does the issue resolve itself when disabling Brave Rewards? No
- Is the issue reproducible on the latest version of Chrome? Yes - This functionality is the same on Chrome.

## Miscellaneous Information:

N/A

**Comment by Ashish0804** [comment]
It would also be awesome(i actually need it for a script) if we can go as low as 1-5% (current lowest is 25%)

**Comment by rebron** [comment]
May need to wait for an upstream fix for this one. @scudder12 Is there an extension for this already available?

**Comment by scudder12** [comment]
Yes, there is one extension that does this with a slider, but it seems to me it should be a part of the browser itself since the need to adjust zoom is becoming more and more common.

**Comment by 2004seraph** [comment]
@scudder12  Would you happen to know of any open source addons that do this?
